### DISCUSSION TITLE
MultiLine field with unlimited length in document libraries

### DISCUSSION THREAD

**Discussion by hoppalazi** [pnp/powershell#4600] [Ideas]
Is there a way to set the property "[UnlimitedLengthInDocumentLibrary](https://learn.microsoft.com/en-us/dotnet/api/microsoft.sharepoint.client.fieldmultilinetext.unlimitedlengthindocumentlibrary?view=sharepoint-csom#microsoft-sharepoint-client-fieldmultilinetext-unlimitedlengthindocumentlibrary)" on a multiline field of a site content type?
There seems to be no way to set this property? Atleast i was not able to find it.

**Comment by waaromikniet** [comment]
Couldn't you just change the fieldXml and update it? I have tested it with this script and it works on my list

```
$field = Get-PnPField -List "UnlimitedLength" -Identity "Samenvatting"
[xml]$XML = $field.SchemaXml
$NodeToEdit = $XML.SelectSingleNode('/')
$NodeToEdit
$attrib = $NodeToEdit.CreateAttribute('UnlimitedLengthInDocumentLibrary')
$attrib.Value = "TRUE"
$NodeToEdit.Field.Attributes.Append($attrib) 
$newFieldXml = $XML.InnerXml;
$newFieldXml
Set-PnPField -List "UnlimitedLength" -Identity "Samenvatting" -Values @{ SchemaXml = $newFieldXml  }
```

**Reply by hoppalazi** [reply]
Well, i shouldn't have forgotten half of the problem description.
The inital problem is that i wanted to change this field property on a site content type.

**Comment by waaromikniet** [comment]
When you have a Site ContentType you need to edit the Site Column. Just change the schemaXml of the sitefield. This works for me:

```
$field = Get-PnPField -Identity "Samenvatting"
[xml]$XML = $field.SchemaXml
$NodeToEdit = $XML.SelectSingleNode('/')
$attrib = $NodeToEdit.CreateAttribute('UnlimitedLengthInDocumentLibrary')
$attrib.Value = "TRUE"
$NodeToEdit.Field.Attributes.Append($attrib) 
$newFieldXml = $XML.InnerXml;
Set-PnPField -Identity "Samenvatting" -Values @{ SchemaXml = $newFieldXml  } -UpdateExistingLists
```

So remove the  `-List "UnlimitedLength"` and add the -UpdateExistingLists to update all lists
